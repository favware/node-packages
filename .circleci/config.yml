version: 2.1

orbs:
  codecov: codecov/codecov@1.0.5

aliases:
  - &checkout
    checkout
  - &store_artifacts
    store_artifacts:
      path: coverage
  - &codecov
    codecov/upload:
      file: coverage/*.json

run_yarn: &run_yarn
  name: Install yarn packages
  command: yarn install --frozen-lockfile

save_git: &save_git
  save_cache:
    name: Save Git Cache
    key: git-code-{{ .Revision }}
    paths:
      - .git
      - ~/.ssh

save_yarn: &save_yarn
  save_cache:
    name: Save Yarn Package Cache
    key: yarn-packages-{{ .Branch }}-{{ checksum "yarn.lock" }}
    paths:
      - ~/.cache/yarn
      - node_modules

restore_git: &restore_git
  restore_cache:
    name: Restore Git Cache
    key: git-code-{{ .Revision }}

restore_yarn: &restore_yarn
  restore_cache:
    name: Restore Yarn Package Cache
    key: yarn-packages-{{ .Branch }}-{{ checksum "yarn.lock" }}

commands:
  greenkeeper_update:
    description: Updates the lockfile if current branch is a greenkeeper branch
    steps:
    - *checkout
    - <<: [*restore_git, *restore_yarn]
    - run:
        name: update lockfile
        command:
          export PATH=$PATH:$(yarn global bin) &&
          yarn global add greenkeeper-lockfile@2 &&
          greenkeeper-lockfile-update
    - run:
        name: Install dependencies
        command: yarn install --frozen-lockfile
    - <<: [*save_yarn, *save_git]
  nodejs:
    description: Primary flow
    steps:
      - *checkout
      - <<: [*restore_git, *restore_yarn]
      - run:
          <<: *run_yarn
      - run:
          name: Lint all libraries
          command: yarn lint
      - run:
          name: Test all libraries
          command: yarn ci:test
      - *store_artifacts
      - *codecov
  greenkeeper_upload:
    description: Uploads the lockfile if current branch is a greenkeeper branch
    steps:
    - *checkout
    - <<: [*restore_git, *restore_yarn]
    - run:
        name: Upload lockfile
        command:
          export PATH=$PATH:$(yarn global bin) &&
          yarn global add greenkeeper-lockfile@2 &&
          greenkeeper-lockfile-update

# List of jobs
jobs:
  lockfile_update:
    docker:
      - image: circleci/node:10
    steps:
      - greenkeeper_update
  node-v10:
    docker:
      - image: circleci/node:10
    steps:
      - nodejs
  node-v11:
    docker:
      - image: circleci/node:11
    steps:
      - nodejs
  node-v12:
    docker:
      - image: circleci/node:12
    steps:
      - nodejs
  lockfile_upload:
    docker:
      - image: circleci/node:10
    steps:
      - greenkeeper_upload

workflows:
  version: 2
  node-multi-build:
    jobs:
      - lockfile_update
      - node-v10:
          requires:
            - lockfile_update
      - node-v11:
          requires:
            - lockfile_update
      - node-v12:
          requires:
            - lockfile_update
      - lockfile_upload:
          requires:
            - node-v10
            - node-v11
            - node-v12

  nightly:
    triggers:
      - schedule:
          cron: 0 2 * * *
          filters:
            branches:
              only:
                - master
    jobs:
      - node-v10
      - node-v11
      - node-v12