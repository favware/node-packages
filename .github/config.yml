pull_request_rules:
  - name: Automatic Depfu merging
    conditions:
      - author=depfu[bot]
      - approved-reviews-by=Favna
      - base=main
      - label~=^Meta:\ Dependencies
    actions:
      merge:
        method: squash
