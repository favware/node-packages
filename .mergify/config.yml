pull_request_rules:
  - name: Automatic Depfu merging
    conditions:
      - approved-reviews-by=Favna
      - base=main
      - label~=^Meta:\ Dependencies
      - title~=Yarn\ dependencies
    actions:
      merge:
        method: squash
