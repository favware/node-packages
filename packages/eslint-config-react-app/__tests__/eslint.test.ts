import config from '../src';

describe('ESLint Config', () => {
  test('should export rules', () => {
    expect(config.root).toBe(true);
    expect(config.extends).toContainEqual('@favware');
    expect(config).toMatchSnapshot();
  });
});
