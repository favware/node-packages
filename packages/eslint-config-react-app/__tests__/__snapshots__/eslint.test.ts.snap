// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ESLint Config should export rules 1`] = `
Object {
  "extends": Array [
    "@favware",
    "plugin:import/typescript",
  ],
  "parserOptions": Object {
    "ecmaFeatures": Object {
      "jsx": true,
    },
    "ecmaVersion": 2019,
    "jsx": true,
    "project": "./tsconfig.eslint.json",
    "sourceType": "module",
  },
  "plugins": Array [
    "import",
    "jsx-a11y",
    "react",
    "react-hooks",
  ],
  "root": true,
  "rules": Object {
    "@typescript-eslint/no-explicit-any": "off",
    "getter-return": "warn",
    "import/first": "error",
    "import/no-amd": "error",
    "import/no-webpack-loader-syntax": "error",
    "jsx-a11y/accessible-emoji": "warn",
    "jsx-a11y/alt-text": "warn",
    "jsx-a11y/anchor-has-content": "warn",
    "jsx-a11y/anchor-is-valid": Array [
      "warn",
      Object {
        "aspects": Array [
          "noHref",
          "invalidHref",
        ],
      },
    ],
    "jsx-a11y/aria-activedescendant-has-tabindex": "warn",
    "jsx-a11y/aria-props": "warn",
    "jsx-a11y/aria-proptypes": "warn",
    "jsx-a11y/aria-role": Array [
      "warn",
      Object {
        "ignoreNonDOM": true,
      },
    ],
    "jsx-a11y/aria-unsupported-elements": "warn",
    "jsx-a11y/heading-has-content": "warn",
    "jsx-a11y/iframe-has-title": "warn",
    "jsx-a11y/img-redundant-alt": "warn",
    "jsx-a11y/no-access-key": "warn",
    "jsx-a11y/no-distracting-elements": "warn",
    "jsx-a11y/no-redundant-roles": "warn",
    "jsx-a11y/role-has-required-aria-props": "warn",
    "jsx-a11y/role-supports-aria-props": "warn",
    "jsx-a11y/scope": "warn",
    "no-extra-label": "warn",
    "no-invalid-this": "off",
    "no-iterator": "warn",
    "no-label-var": "warn",
    "no-labels": Array [
      "warn",
      Object {
        "allowLoop": true,
        "allowSwitch": true,
      },
    ],
    "no-lone-blocks": "warn",
    "no-loop-func": "warn",
    "no-mixed-operators": Array [
      "warn",
      Object {
        "allowSamePrecedence": false,
        "groups": Array [
          Array [
            "&",
            "|",
            "^",
            "~",
            "<<",
            ">>",
            ">>>",
          ],
          Array [
            "==",
            "!=",
            "===",
            "!==",
            ">",
            ">=",
            "<",
            "<=",
          ],
          Array [
            "&&",
            "||",
          ],
          Array [
            "in",
            "instanceof",
          ],
        ],
      },
    ],
    "no-multi-str": "warn",
    "no-restricted-properties": Array [
      "error",
      Object {
        "message": "Please use import() instead. More info: https://facebook.github.io/create-react-app/docs/code-splitting",
        "object": "require",
        "property": "ensure",
      },
      Object {
        "message": "Please use import() instead. More info: https://facebook.github.io/create-react-app/docs/code-splitting",
        "object": "System",
        "property": "import",
      },
    ],
    "react-hooks/exhaustive-deps": "warn",
    "react-hooks/rules-of-hooks": "error",
    "react/forbid-foreign-prop-types": Array [
      "warn",
      Object {
        "allowInPropTypes": true,
      },
    ],
    "react/jsx-no-comment-textnodes": "warn",
    "react/jsx-no-duplicate-props": "warn",
    "react/jsx-no-target-blank": "warn",
    "react/jsx-no-undef": "error",
    "react/jsx-pascal-case": Array [
      "warn",
      Object {
        "allowAllCaps": true,
        "ignore": Array [],
      },
    ],
    "react/jsx-uses-react": "warn",
    "react/jsx-uses-vars": "warn",
    "react/no-access-state-in-setstate": "warn",
    "react/no-children-prop": "warn",
    "react/no-danger": "error",
    "react/no-danger-with-children": "warn",
    "react/no-direct-mutation-state": "warn",
    "react/no-is-mounted": "warn",
    "react/no-this-in-sfc": "warn",
    "react/no-typos": "error",
    "react/react-in-jsx-scope": "error",
    "react/require-render-return": "error",
    "react/style-prop-object": "warn",
    "react/void-dom-elements-no-children": "warn",
    "rest-spread-spacing": Array [
      "warn",
      "never",
    ],
  },
  "settings": Object {
    "react": Object {
      "version": "detect",
    },
  },
}
`;
